// react-native-get-location/dist/index.js
"use strict";
/**
 * MIT License
 *
 * Copyright (c) 2019 Douglas Nassif Roma Junior
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.isLocationError = exports.LocationError = void 0;
const react_native_1 = require("react-native");
const utils_1 = require("./utils");
const LocationError_1 = __importStar(require("./LocationError"));
var LocationError_2 = require("./LocationError");
Object.defineProperty(exports, "LocationError", { enumerable: true, get: function () { return __importDefault(LocationError_2).default; } });
Object.defineProperty(exports, "isLocationError", { enumerable: true, get: function () { return LocationError_2.isLocationError; } });
const { OS } = react_native_1.Platform;
const { ReactNativeGetLocation } = react_native_1.NativeModules;
const DEFAULT_OPTIONS = {
    enableHighAccuracy: false,
    timeout: 60000,
};
const GetLocation = {
    async getCurrentPosition(options = DEFAULT_OPTIONS) {
        const opts = { ...DEFAULT_OPTIONS, ...options };
        if (OS === 'android') {
            await (0, utils_1.requestAndroidPermission)(opts.enableHighAccuracy, opts.rationale);
        }
        try {
            return ReactNativeGetLocation.getCurrentPosition(opts);
        }
        catch (error) {
            if ((0, LocationError_1.isLocationError)(error)) {
                throw new LocationError_1.default(error.code, error.message);
            }
            throw error;
        }
    },
    openSettings() {
        return react_native_1.Linking.openSettings();
    },
};
exports.default = GetLocation;
